static/js/src/appliance.js

```diff
@@ -0,0 +1,25 @@
+if (typeof appliance !== "undefined") {
+  throw TypeError("Namespace 'appliance' not available");
+}
+
+var appliance = {};
+
+appliance.updateURL = function(linkElement, selectElement) {
+  linkElement.href = window.location.pathname + '/' + selectElement.value;
+}
+
+appliance.downloadFormsInit = function(downloadForms) {
+  if (downloadForms) {
+    downloadForms.forEach(function(container) {
+      const linkElement = container.querySelector('.js-download-link');
+      const selectElement = container.querySelector('.js-platform-select');
+      if (linkElement && selectElement) {
+        selectElement.addEventListener('change', function() {
+          appliance.updateURL(linkElement, selectElement);
+        });
+      }
+    });
+  }
+}
+
+appliance.downloadFormsInit(document.querySelectorAll(".js-download"));
```

Comment:
We shouldn't require js to download something.

Reply:
Agreed. I was going to add a no-JS fallback but was expecting some discussion around this solution.